core/unit_test/TestComplex.hpp

```diff
@@ -688,6 +687,43 @@ TEST(TEST_CATEGORY, complex_structured_bindings) {
   test.testit();
 }
 
+#define CHECK_COMPLEX(_value_, _real_, _imag_) \
+  if (_value_.real() != _real_) return false;  \
+  if (_value_.imag() != _imag_) return false;
+
+constexpr bool construction_is_constexpr() {
+  [[maybe_unused]] const Kokkos::complex<double> from_single{1.2};
+  [[maybe_unused]] const Kokkos::complex<double> from_both{1.2, 3.4};
+  [[maybe_unused]] const Kokkos::complex<double> from_none{};
```

Comment:
I would have dropped all the maybe_unused attributes and just cast to void in the CHECK_COMPLEX macro.